### PR TITLE
LLVM 3.4 + 3.8 Refactor

### PR DESCRIPTION
**Pull request by parkre**
• 3.8 LLVM plugin code is reformatted similar to 3.4
• `llvm_binary_legacy.hpp` is updated (i.e. the same as upstream `llvm_binary.hpp`)

### CODE REVIEW

File: lib/bap_ida/bap_ida_config.ml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+let ida_path = "#undefined"
+let is_headless = false
```
Comment (ivg) [inline lib/bap_ida/bap_ida_config.ml:2]: This file is autogenerated, it shouldn't be here or in any other history

File: plugins/llvm/llvm_binary.cpp [hunk 2]
```diff
@@ -46,59 +46,59 @@ extern "C" {
 
 
     const char* segment_name(const seg::segment* s) {
-        return s->name().c_str();
+        return s->name.c_str();
```
Comment (ivg) [inline plugins/llvm/llvm_binary.cpp:49]: sync your master with the upstream, there is no need to review changes that are already in the upstream.

Comment (parkre) [inline plugins/llvm/llvm_binary.cpp:49]: Yeah My fault, sorry

**Comment by ivg** [comment]
moved this review into #8, please close this PR.

**Pull request closed without merging**